superscripts.rkt

```diff
@@ -114,6 +115,16 @@
     [(syntax? e) (datum->syntax e (add-scopes (syntax-e e)) e e)]
     [(pair? e) (cons (add-scopes (car e))
                      (add-scopes (cdr e)))]
+    [(box? e)
+     (box-immutable (add-scopes (unbox e)))]
+    [(vector? e)
+     (apply vector-immutable (map add-scopes (vector->list e)))]
+    [(hash? e)
+     (for/fold ([e e]) ([k (in-hash-keys e)])
```

Comment:
Out of curiosity, why use `for/fold` instead of `hash-map`? Does this have something to do with the identity of the (possibly mutable) hash object?

Reply:
Because `hash-map` returns a list, not a hash.

Reply:
Oh yeah, of course. I'm always surprised when I realise that :) .

Reply:
We also considered `for/hash`, but then realized it doesn't preserve the "equal" operation as in whether it's an `eq?`, `eqv?`, or `equal?` based hash, while `hash-update` does preserve that.